Sources/TuistKit/Services/Inspect/InspectBuildService.swift

```diff
@@ -0,0 +1,197 @@
+
+        if environment.tuistVariables["TUIST_INSPECT_BUILD_WAIT"] != "YES",
+           environment.workspacePath != nil
+        {
+            var environment = ProcessInfo.processInfo.environment
+            environment["TUIST_INSPECT_BUILD_WAIT"] = "YES"
+            // We don't want to prolongue the build action for analytics reasons.
+            // Additionally, the `.xcactivitylog` might not be immediately available.
+            // To resolve both issues, we run the `inspect build` command again when run from a post action (recognized by the
+            // presense of the `workspacePath` environment variable).
+            // We pass the `TUIST_INSPECT_BUILD_WAIT` environment variable to the new process, so we do actually upload the build
+            // analytics to the server.
+            try backgroundProcessRunner.runInBackground(
+                [executablePath, "inspect", "build"],
+                environment: environment
+            )
+            return
+        }
+        let projectPath = try await projectPath(path)
+        let buildLogsPath = try derivedDataLocator.locate(for: projectPath)
+            .appending(components: "Logs", "Build")
+        let logManifestPlistPath = buildLogsPath.appending(component: "LogStoreManifest.plist")
+
+        guard try await fileSystem.exists(logManifestPlistPath)
+        else {
+            throw InspectBuildServiceError.noBuildLogFound(buildLogsPath: buildLogsPath, projectPath: projectPath)
+        }
+        let xcactivityLog = try await latestXCActivityLog(
+            logManifestPlistPath: logManifestPlistPath,
+            buildLogsPath: buildLogsPath,
+            projectPath: projectPath,
+            referenceDate: referenceDate
+        )
+        try await createBuild(
+            for: xcactivityLog,
+            projectPath: projectPath
+        )
+    }
+
+    private func createBuild(
+        for xcactivityLog: XCActivityLog,
+        projectPath: AbsolutePath
+    ) async throws {
+        let config = try await configLoader
+            .loadConfig(path: projectPath)
+        guard let fullHandle = config.fullHandle else { throw InspectBuildServiceError.missingFullHandle }
+        try await createBuildService.createBuild(
+            fullHandle: fullHandle,
+            serverURL: config.url,
+            id: xcactivityLog.mainSection.uniqueIdentifier,
+            duration: Int(xcactivityLog.mainSection.timeStoppedRecording * 1000) -
+                Int(xcactivityLog.mainSection.timeStartedRecording * 1000),
+            isCI: ciChecker.isCI(),
+            modelIdentifier: machineEnvironment.modelIdentifier(),
+            macOSVersion: machineEnvironment.macOSVersion,
+            scheme: environment.schemeName,
+            xcodeVersion: try await xcodeBuildController.version()?.description
+        )
+        ServiceContext.current?.ui?.success(
+            .alert(
+                "Uploaded a build to the server."
+            )
+        )
```

Comment:
The idea here is that you'll be able to go to the detail page of the build and get something similar to the xclogparser reporter, but nicer:
![image](https://github.com/user-attachments/assets/0a7b1a8c-a3c0-496c-b328-57ed75542290)

For now, I don't expect developers to run this directly a lot – it should mostly be run in a scheme's post action.

Reply:
It looks like a great place for integration with [Build Graph](https://github.com/akaDuality/build-graph-for-xcode)